openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -23,202 +23,114 @@
+
+
 class PhylopicDataIngester(ProviderDataIngester):
     delay = 5
-    host = "http://phylopic.org"
-    # Use "base_endpoint" since class's "endpoint" parameter gets defined as a property
-    base_endpoint = f"{host}/api/a/image"
+    host = "https://www.phylopic.org"
+    endpoint = "https://api.phylopic.org/images"
     providers = {constants.IMAGE: prov.PHYLOPIC_DEFAULT_PROVIDER}
-    batch_limit = 25
 
-    @property
-    def endpoint(self) -> str:
-        """
-        Due to the way this DAG is run (via a dated range), **only one request is ever
-        issued** to retrieve all updated IDs. As such, it will only
-        return one endpoint.
-        """
-        list_endpoint = f"{self.base_endpoint}/list"
-        # Process for a given date
-        end_date = (date.fromisoformat(self.date) + timedelta(days=1)).isoformat()
-        # Get a list of objects uploaded/updated within a date range
-        # http://phylopic.org/api/#method-image-time-range
-        endpoint = f"{list_endpoint}/modified/{self.date}/{end_date}"
-        logger.info(f"Constructed endpoint: {endpoint}")
-        return endpoint
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.current_page = 0
+        self.total_pages = 0
+        self.build_param = 0
+
+    def ingest_records(self):
+        self._get_initial_query_params()
+        super().ingest_records()
+
+    def _get_initial_query_params(self) -> None:
+        """Get the required `build` param from the API and set the total pages."""
+        resp = self.get_response_json(query_params={})
+        if not resp:
+            raise Exception("No response from Phylopic API.")
+        self.build_param = resp.get("build")
+        self.total_pages = resp.get("totalPages")
+        logger.info(
+            f"Total items to fetch: {resp.get('totalItems')}. "
+            f"Total pages: {self.total_pages}"
+        )
 
     def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
-        """Noop since the range is determined by the endpoint property."""
-        return {}
+        if prev_query_params is not None:
+            self.current_page += 1
 
-    def get_should_continue(self, response_json):
-        """
-        Override for upstream "return True". This DAG will only ever make 1 query so
-        they should not continue to loop.
-        """
-        return False
+        return {
+            "build": self.build_param,
+            "page": self.current_page,
+            "embed_items": "true",
+        }
 
-    @staticmethod
-    def _get_response_data(response_json) -> dict | list | None:
-        """Intermediate method for pulling out results from a Phylopic API request."""
-        if response_json and response_json.get("success") is True:
-            return response_json.get("result")
+    def get_should_continue(self, response_json):
+        return self.current_page < self.total_pages
```

Comment:
```suggestion
        return self.current_page <= self.total_pages
```
This DAG ran really well until the last page, when the last requested page returned 404:
```
[2023-03-27, 15:45:24 +03] {requester.py:78} WARNING - Unable to request URL: https://api.phylopic.org/images?build=196&page=146&embed_items=true  Status code: 404
[2023-03-27, 15:45:24 +03] {requester.py:148} WARNING - Bad response_json:  None
[2023-03-27, 15:45:24 +03] {requester.py:149} WARNING - Retrying:
_get_response_json(
    https://api.phylopic.org/images,
    {'build': 196, 'page': 146, 'embed_items': 'true'},
    retries=-1)
[2023-03-27, 15:45:24 +03] {requester.py:134} ERROR - No retries remaining.  Failure.
[2023-03-27, 15:45:24 +03] {media.py:230} INFO - Writing 74 lines from buffer to disk.
[2023-03-27, 15:45:24 +03] {provider_data_ingester.py:501} INFO - Committed 6974 records
[2023-03-27, 15:45:24 +03] {taskinstance.py:1768} ERROR - Task failed with exception
providers.provider_api_scripts.provider_data_ingester.IngestionError: Retries exceeded
query_params: {"build": 196, "page": 146, "embed_items": "true"}
```

The `initial_request.json`(tests/dags/providers/provider_api_scripts/resources/phylopic/initial_request.json) seems to suggest that we should use `<=` or even `==` here, because even thought the totalPages is 145, the lastPage is 144.
```
  "lastPage": {
    "href": "/images?build=194&page=144"
  },
  "totalPages": 145
```